active/0000-plaforms.md

```diff
@@ -0,0 +1,328 @@
+- Start Date: (fill me in with today's date, YYYY-MM-DD)
+- RFC PR: (leave this empty)
+- ember-cli Issue: (leave this empty)
+
+# Summary
+
+The ember-cli platform flag treats platforms as build targets, and gives you a
+workflow for producing builds customized to a platform, as well as streamlined tooling,
+enhanced resuability, and better separation of concerns for applications that
+target more than one platform.  Platforms are roughly equivalent to web views,
+or a shell which provides a web view, such as a browser.
+
+# Motivation
+
+There is a growing trend of shipping Ember applications inside of WebView shells to
+provide installable cross-platform applications and to gain access to native features.
+
+Currently, each platform has a homegrown solution for managing build/serve/test/deploy
+processes.  We could reduce the surface area of commands these platforms expose and
+ease the building of addons specific to a platform by enabling ember-cli to understand
+the existence of platforms and how various assets relate to them.
+
+# Detailed design
+
+## Usage
+
+These are the docs for using `ember-cli-platforms` with an installed platform plugin.
+For the docs on creating a platform plugin, [skip ahead](./README.md#creating-plugins).
+
+**Create a platform.**
+```cli
+ember platform <type> [-t "<name>"]
+```
+
+Generates a new platform of `<type>` with an optional `<name>` parameter used when more than
+one instance of a platform is desired.
+
+With Cordova this might look like the following.
+
+Default setup (one project)
+```cli
+ember platform cordova
+```
+
+Advanced setup (multiple projects)
+```cli
+ember platform cordova -t "android"
+ember platform cordova -t "ios"
+```
+
+For all commands, `platform` can be abbreviated as `p`, and
+commands that mirror `ember-cli` commands preserve the `ember-cli`
+shorthands for those methods as well.
+
+**Serve a specific platform.**
+```cli
+ember s -p "<type>[-<name>]"
+```
+
+This is shorthand for:
+```cli
+ember serve --platform="<type>" [--target="<name>"] --environment="development"
+```
+
+**Build a specific platform.**
+```cli
+ember b -p "<type>[-<name>]"
+```
+
+**Test a specific platform.**
+```cli
+ember t -p "<type>[-<name>]"
+```
+
+**Deploy a specific platform.**
+```cli
+ember d <deployTarget> -p "<type>[-<name>]" 
+```
+
+This is short hand for:
+```cli
+ember deploy --deployTarget="<deployTarget>" --platform="<type>" [--target="<name>"] --environment="development"
+```
+
+
+### Platform Specific Code
+
+Platforms would leverage in-repo-addon semantics and Broccoli plugins to enable
+sprinkling of platform specific code into an app as needed with "platform addons".
+
+A platform addon is generated automatically for each platform created with `ember platform`.
+
+```
+<project>
+  /lib
+    /<type>[-<name>]
+```
+
+These "addons" function differently from normal addons, and would be prefixed with `-platform-`.
```

Comment:
They don't necessarily have to have a config file, I think relying on a potentially volatile (and easily deletable) configuration file is going to be trickier than distinguishing between in-repo addons and in-repo platforms in this way.